semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/NodeMethods.scala

```diff
@@ -6,10 +6,17 @@ import io.shiftleft.semanticcpg.language.*
 
 class NodeMethods(val node: AbstractNode) extends AnyVal with NodeExtension {
 
+  @deprecated("Prefer the .loc node step which lazily computes location information.")
   def location(implicit finder: NodeExtensionFinder): NewLocation =
     node match {
       case storedNode: StoredNode => LocationCreator(storedNode)
       case _                      => LocationCreator.emptyLocation("", None)
     }
 
+  def loc: Loc = {
+    node match {
+      case storedNode: StoredNode => Loc(storedNode)
+    }
```

Comment:
In order to make this a replacement for `def location` please also add the empty location equivalent. Otherwise we will get match exceptions.

Reply:
Added, see `Location.apply` and `LocationCreator` trait